### ISSUE TITLE
`go get -u` is failing

### ISSUE THREAD

**Issue by markfuller** [lyraproj/lyra#126]
`go get -u` is failing

```sh
➜  lyra git:(master) ✗ go test ./...
go: finding sigs.k8s.io/structured-merge-diff/merge latest
go: finding k8s.io/api/admissionregistration/v1alpha1 latest
go: finding sigs.k8s.io/structured-merge-diff latest
go: finding k8s.io/api/admissionregistration latest
go: finding k8s.io/api latest
../../go/pkg/mod/k8s.io/client-go@v10.0.0+incompatible/kubernetes/scheme/register.go:22:2: unknown import path "k8s.io/api/admissionregistration/v1alpha1": cannot find module providing package k8s.io/api/admissionregistration/v1alpha1
```

Workaround.

To update lyraproj mods other than lyra-operator:

```
go get -u github.com/lyraproj/puppet...
go get -u github.com/lyraproj/s...
go get -u github.com/lyraproj/hiera
go get -u github.com/lyraproj/issue
go get -u github.com/lyraproj/wfe
```

To update lyra-operator:

(in a clean branch with nothing modified)

* Run `go get -u`
* Copy only the go.mod line for github.com/lyraproj/lyra-operator
* Run `git checkout .`
* Paste the copied line over the go.mod line for github.com/lyraproj/lyra-operator
* Run make or similar

**Comment by markfuller** [comment]
tl;dr using `go get github.com/lyraproj/...` without the `-u` should work well.

The behaviour of `-u` is "Update all direct and indirect dependencies to latest minor or patch upgrades" ([ref](https://github.com/golang/go/wiki/Modules#daily-workflow)).  Dependencies of lyra-operator are typically non-mods and have [Pseudo versions](https://golang.org/cmd/go/#hdr-Pseudo_versions) e.g. `v0.0.0-20190128043916-71123fcbb8fe`, i.e. no semantic versions, meaning that the latest will always be pulled if updated.

The problem here is that where C depends on B and B depends on A, and the latest version of A is incompatible with the "preferred" version of B.  In this case B is k8s.io/client-go and because there is a semantically well-named (v10.0.0) release in github (https://github.com/kubernetes/client-go/releases/tag/v10.0.0), it is considered to be preferable to later commits.  So `go get -u` will pull the latest (master) of `k8s.io/api` but not the latest of `k8s.io/client-go`

We could use a replace as follows (i.e. specify that we wish to use a specific, later, version of B).  This survives a `go get -u` (unlike updating the reference of k8s.io/client-go), but we could end up maintaining a list here.

```
replace k8s.io/client-go => k8s.io/client-go v2.0.0-alpha.0.0.20190215080940-fc20e1abff02+incompatible
```

Easier than this, we could simply run `go get github.com/lyraproj/...` which will update the lyraproj dependencies but only update their dependencies if specified by their `go.mod` files.

Aside: the following command should `go get` all direct dependencies.

```
go list -m -f '{{ .Path }} {{ .Indirect }}' all | grep " false" | awk '{print $1}' | xargs go get
```

but is prone to a higher risk of failure.  Currently, for example, we need to exclude terraform for this to work:

```
go list -m -f '{{ .Path }} {{ .Indirect }}' all | grep " false" | grep -v "github.com/hashicorp/terraform" | awk '{print $1}' | xargs go get
```
